lib/common/kind-variant.h

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
```

Comment:
Oops!  I wrote this code and then ended up not using it anywhere.  Please ignore it until it disappears from this review.